src/overlays/actors/ovl_En_Rz/z_en_rz.c

```diff
@@ -42,80 +61,646 @@ const ActorInit En_Rz_InitVars = {
+    };
+    static AnimationHeader* sMarillaAnimations[] = {
+        &gRosaSistersStandingAnim,   &gRosaSistersStandingAnim, &gRosaSistersWalkingAnim, &gRosaSistersSittingAnim,
+        &gRosaSistersApplaudingAnim, &gRosaSistersOnKneesAnim,  &gRosaSistersDancingAnim,
+    };
+    static LinkAnimationHeader* sLinkAnimations[] = {
+        &gPlayerAnim_link_normal_wait_free,
+        &gPlayerAnim_alink_dance_loop,
+    };
+    f32 endFrame;
+    AnimationHeader** animationPtr;
+
+    if (EN_RZ_GET_SISTER(&this->actor) == EN_RZ_JUDO) {
+        animationPtr = sJudoAnimations;
+    } else { // EN_RZ_MARILLA
+        animationPtr = sMarillaAnimations;
+    }
+
+    if ((animIndex >= EN_RZ_ANIM_THINKING) && (animIndex < EN_RZ_ANIM_MAX) &&
+        ((animIndex != this->animIndex) || (animMode != ANIMMODE_LOOP))) {
+        if (animIndex >= ARRAY_COUNT(sJudoAnimations)) {
+            endFrame = Animation_GetLastFrame(sLinkAnimations[animIndex - ARRAY_COUNT(sJudoAnimations)]);
+            if (animMode == ANIMMODE_LOOP) {
+                LinkAnimation_Change(play, &this->skelAnime, sLinkAnimations[animIndex - ARRAY_COUNT(sJudoAnimations)],
+                                     2.0f / 3.0f, 0.0f, endFrame, ANIMMODE_LOOP, morphFrames);
+            } else {
+                LinkAnimation_Change(play, &this->skelAnime, sLinkAnimations[animIndex - ARRAY_COUNT(sJudoAnimations)],
+                                     2.0f / 3.0f, 0.0f, endFrame, ANIMMODE_LOOP, morphFrames);
+            }
+        } else {
+            Animation_Change(&this->skelAnime, animationPtr[animIndex], 1.0f, 0.0f,
+                             Animation_GetLastFrame(animationPtr[animIndex]), animMode, morphFrames);
+        }
+
+        this->animIndex = animIndex;
+    }
+}
+
+void func_80BFB9E4(PlayState* play, EnRz* this, s16 animIndex) {
+    EnRz_ChangeAnim(play, this, animIndex, ANIMMODE_LOOP, -5.0f);
+}
+
+void func_80BFBA1C(PlayState* play, EnRz* this, s16 animIndex) {
+    EnRz_ChangeAnim(play, this, animIndex, ANIMMODE_LOOP, 0.0f);
+}
+
+s32 EnRz_SetupPath(EnRz* this, PlayState* play) {
+    if (EN_RZ_GET_PATH(&this->actor) != 0x3F) {
+        this->path = &play->setupPathList[EN_RZ_GET_PATH(&this->actor)];
+        if (this->path != NULL) {
+            Path* path = this->path;
+            Vec3s* points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+            f32 diffX = points->x - this->actor.world.pos.x;
+            f32 diffY = points->z - this->actor.world.pos.z;
+
+            if ((SQ(diffX) + SQ(diffY)) < SQ(10.0f)) {
+                this->curPointIndex = 0;
+                this->stateFlags &= ~EN_RZ_STATE_1;
+            } else {
+                this->curPointIndex = path->count - 1;
+                this->stateFlags |= EN_RZ_STATE_1;
+                return true;
+            }
+        }
+    } else {
+        this->path = NULL;
+        this->curPointIndex = 0;
+    }
+    return false;
+}
+
+EnRzGetPathStatus EnRz_PathStatus(EnRz* this) {
```

Comment:
Oh, my comment was referring to the function name, not the enum
`EnRz_GetPathStatus`